src/index.ts

```diff
@@ -1,58 +1,68 @@
 import axios, { AxiosRequestHeaders } from "axios";
+import {
+  SdkEventType,
+  GtmhubSdkDc,
+  Setting,
+  sdkEventTypeCollection,
+} from "./models";
+const sdkVersion = require("../package.json").version;
+
+const DEFAULT_GTMUB_SDK_DC = "eu";
 
-type EventType = "linkIssue" | "getCurrentItem" | "updateCurrentItem" | GeneralEvents | PluginEvents | MetricEvents | GoalEvents | TaskEvents | SessionEvents | UserEvents | AccountEvents | TeamEvents;
-type MetricEvents = "getMetrics" | "getMetric" | "createMetric" | "updateMetric" | "deleteMetric" | "checkinMetric" | "patchMetricComment" | "createMetricReaction" | "deleteMetricReaction" | "deleteMetricSnapshot" | "onMetricUpdated";
-type GoalEvents = "getGoal" | "getGoals" | "createGoal" | "updateGoal" | "deleteGoal" | "onGoalUpdated";
-type TaskEvents = "getTask" | "getTasks" | "createTask" | "updateTask" | "deleteTask";
-type UserEvents = "getUser" | "getUsers" | "getCurrentUser" | "createUser" | "updateUser" | "deleteUser";
-type AccountEvents = "getAccountId" | "getAccountUrl" | "getAccountSettings" | "getCustomFields";
-type TeamEvents = "getTeam" | "getTeams" | "createTeam" | "updateTeam" | "deleteTeam";
-type PluginEvents = "getSetting" | "getSettings";
-type SessionEvents = "getSession";
-type GeneralEvents = "getDc" | "resize" | "getAssigneeById" | "getAssigneesByIds";
-type Setting = { key: string; value: string };
-
-type dc = "us" | "eu" | "staging";
 class Gtmhub {
-  pluginId = "";
-  dc;
+  private pluginId: string = "";
+  private dc: GtmhubSdkDc = DEFAULT_GTMUB_SDK_DC;
```

Comment:
no need to type these varaibles when assigning a value to them, they should inherit the type from the value when being assigned